### PR TITLE
Feature: Remove a custom app

### PR DESCRIPTION
**Pull request by mikhailxyz**
## What it solves

- Adds a button to remove a custom app
- Fixes the test for adding a custom app

## How to test it
Add a custom app, remove it, and refresh the app. The custom app shouldn't be there anymore

## Analytics changes
None

## Screenshots
image
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "lint": "tsc && next lint",
     "lint:fix": "yarn lint --fix",
     "prettier": "prettier -w \"{src,cypress,mocks,scripts}/**/*.{ts,tsx,css,js}\"",
-    "test": "DEBUG_PRINT_LIMIT=15000 jest",
+    "test": "DEBUG_PRINT_LIMIT=30000 jest",
```
Comment (mikhailxyz) [inline package.json:16]: 📈 📈 📈 🚀

File: src/components/safe-apps/RemoveCustomAppModal.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  app: SafeAppData
+  onClose: () => void
+  onConfirm: (appId: number) => void
+}
+
+const RemoveCustomAppModal = ({ open, onClose, onConfirm, app }: Props) => (
+  <ModalDialog open={open} onClose={onClose} dialogTitle="Confirm app removal">
+    <DialogContent>
+      <Typography variant="h6" sx={{ pt: 3 }}>
+        Are you sure you want to remove <b>{app.name}</b> app?
```
Comment (katspaugh) [inline src/components/safe-apps/RemoveCustomAppModal.tsx:17]:
```suggestion
        Are you sure you want to remove the <b>{app.name}</b> app?
```

File: src/hooks/safe-apps/useRemoveAppModal.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { SafeAppData } from '@gnosis.pm/safe-react-gateway-sdk'
+
+type ModalState =
+  | {
+      isOpen: false
+      app: null
+    }
+  | {
+      isOpen: true
+      app: SafeAppData
```
Comment (katspaugh) [inline src/hooks/safe-apps/useRemoveAppModal.ts:11]: Perhaps something cooler can be done here with [conditional types](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html)?

Comment (mikhailxyz) [reply src/hooks/safe-apps/useRemoveAppModal.ts:11]: This type doesn't take an input type, so I don't know if it's applicable here. In general, yeah I wonder if there's a better way to do this

Comment (mikhailxyz) [reply src/hooks/safe-apps/useRemoveAppModal.ts:11]: the old typescript handbook had a somewhat similar example: https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions

But I can't find any reference to this pattern in the new docs

Comment (mikhailxyz) [reply src/hooks/safe-apps/useRemoveAppModal.ts:11]: I've just discovered that the pattern I've used here is called `Discriminating Union` 😎

**Review by katspaugh — APPROVED** [review]
Muy bien

**Comment by mikhailxyz** [comment]
gracias senor

**Pull request merged**